CI linter fails on NewDeleteLeaks diagnostics

## Problem
The GitHub Actions Lint job fails on main with clang-tidy 15 diagnostics from clang-analyzer-cplusplus.NewDeleteLeaks.

Run: https://github.com/antgroup/vsag/actions/runs/29188929956/job/86640063910
Head SHA: 0e3e6ca5f300ed5d845a1565b55872ad3d9483ee

## Diagnostics

```text
src/storage/serialization.cpp:127:1: error: Potential leak of memory pointed to by field _M_pi [clang-analyzer-cplusplus.NewDeleteLeaks,-warnings-as-errors]
1 warning treated as error
```

```text
src/factory/factory.cpp:91:1: error: Potential leak of memory pointed to by field _M_pi [clang-analyzer-cplusplus.NewDeleteLeaks,-warnings-as-errors]
1 warning treated as error
```

## Impact
The linter workflow exits with make lint failure, so CI remains red even after other clang-tidy diagnostics are fixed.

## Expected
make lint should pass under the CI clang-tidy 15 environment.

<!-- vsag-github-agent:ci-failure:start -->
## Automated CI failure occurrences

Observed **27** times. This table is maintained by @vsag-bot.

| Time (UTC) | Branch | Commit | Workflow | Failure |
| --- | --- | --- | --- | --- |
| 2026-08-21T03:07:37.000Z | main (1.1) | [`4dab714`](https://github.com/antgroup/vsag/commit/4dab714891502a5f89a68e1850e80eba54cd0050) | Lint | [run #32439920792 attempt 1](https://github.com/antgroup/vsag/actions/runs/32439920792) |
| 2026-08-21T04:27:25.000Z | main (1.1) | [`ad4182e`](https://github.com/antgroup/vsag/commit/ad4182ec1c8833949c0120c1e8ea5a147d9e1b17) | Lint | [run #32444189312 attempt 1](https://github.com/antgroup/vsag/actions/runs/32444189312) |
| 2026-08-21T09:05:10.000Z | main (1.1) | [`9a9c9a8`](https://github.com/antgroup/vsag/commit/9a9c9a83ed0e9b312628f773fb4ddbd9f4d6d07d) | Lint | [run #32462381253 attempt 1](https://github.com/antgroup/vsag/actions/runs/32462381253) |
| 2026-08-21T09:50:13.000Z | main (1.1) | [`b9d8ba0`](https://github.com/antgroup/vsag/commit/b9d8ba04b0337ae430de8777cddc5fedd70d8814) | Lint | [run #32467253679 attempt 1](https://github.com/antgroup/vsag/actions/runs/32467253679) |
| 2026-08-24T04:07:50.000Z | main (1.1) | [`c7b6a54`](https://github.com/antgroup/vsag/commit/c7b6a54e217ce882ca94702ab85b06bb6b66641d) | Lint | [run #32686441917 attempt 1](https://github.com/antgroup/vsag/actions/runs/32686441917) |
| 2026-08-24T12:21:11.000Z | main (1.1) | [`28ac93f`](https://github.com/antgroup/vsag/commit/28ac93f4e0d681af7081386957cbae108b1e59c7) | Lint | [run #32722627520 attempt 1](https://github.com/antgroup/vsag/actions/runs/32722627520) |
| 2026-08-25T08:45:28.000Z | main (1.1) | [`ae4865c`](https://github.com/antgroup/vsag/commit/ae4865c942e018dfcfcb49255fed3546f7d8dc51) | Lint | [run #32824855444 attempt 1](https://github.com/antgroup/vsag/actions/runs/32824855444) |
| 2026-08-26T04:33:38.000Z | main (1.1) | [`06ab479`](https://github.com/antgroup/vsag/commit/06ab479ec638a6f07df8dbd5cd21dfad221e2fe0) | Lint | [run #32927790688 attempt 1](https://github.com/antgroup/vsag/actions/runs/32927790688) |
| 2026-08-26T07:25:20.000Z | main (1.1) | [`3e55243`](https://github.com/antgroup/vsag/commit/3e552431404e53ef681b2a819b20ecc7137f6544) | Lint | [run #32939192229 attempt 1](https://github.com/antgroup/vsag/actions/runs/32939192229) |
| 2026-08-26T07:33:21.000Z | main (1.1) | [`a6aa383`](https://github.com/antgroup/vsag/commit/a6aa3839eb38ec42a0f4aaed429aa06abf227693) | Lint | [run #32939221105 attempt 1](https://github.com/antgroup/vsag/actions/runs/32939221105) |
| 2026-08-26T11:38:49.000Z | main (1.1) | [`b3fd925`](https://github.com/antgroup/vsag/commit/b3fd925e58ff745bff02e0904c2fcf46ca12ee2e) | Lint | [run #32961064903 attempt 1](https://github.com/antgroup/vsag/actions/runs/32961064903) |
| 2026-08-27T05:35:47.000Z | main (1.1) | [`c9a9f16`](https://github.com/antgroup/vsag/commit/c9a9f164ae909ee9b46a2b1dc7881bd4e587a903) | Lint | [run #33040828803 attempt 1](https://github.com/antgroup/vsag/actions/runs/33040828803) |
| 2026-08-31T06:39:39.000Z | main (1.1) | [`ca1f72c`](https://github.com/antgroup/vsag/commit/ca1f72cbfa37bde7fad10970e8b1652ce5893b50) | Lint | [run #33363081672 attempt 1](https://github.com/antgroup/vsag/actions/runs/33363081672) |
| 2026-08-31T09:00:45.000Z | main (1.1) | [`d9af013`](https://github.com/antgroup/vsag/commit/d9af0133de15ba136e826a1d6408c0ddbee0658c) | Lint | [run #33371819401 attempt 1](https://github.com/antgroup/vsag/actions/runs/33371819401) |
| 2026-09-01T03:04:04.000Z | main (1.1) | [`158ad8c`](https://github.com/antgroup/vsag/commit/158ad8c78dedfd5ce1e885bd63223cd1a579e28f) | Lint | [run #33462789022 attempt 1](https://github.com/antgroup/vsag/actions/runs/33462789022) |
| 2026-09-02T03:15:53.000Z | main (1.1) | [`deb1987`](https://github.com/antgroup/vsag/commit/deb1987eda97dff25e2152185d95d02e4bb9ea6e) | Lint | [run #33583851118 attempt 1](https://github.com/antgroup/vsag/actions/runs/33583851118) |
| 2026-09-02T03:23:10.000Z | main (1.1) | [`902ed0b`](https://github.com/antgroup/vsag/commit/902ed0b65cc863e4ff7a6309f1baccc23022151f) | Lint | [run #33584163107 attempt 1](https://github.com/antgroup/vsag/actions/runs/33584163107) |
| 2026-09-02T03:54:01.000Z | main (1.1) | [`c591f1e`](https://github.com/antgroup/vsag/commit/c591f1ea489e479528b7403b632932672f6569fb) | Lint | [run #33585387745 attempt 1](https://github.com/antgroup/vsag/actions/runs/33585387745) |
| 2026-09-02T04:04:32.000Z | main (1.1) | [`6c0a3d7`](https://github.com/antgroup/vsag/commit/6c0a3d77a319dc1ba6062f3b6b901625d0fb8e1d) | Lint | [run #33587404487 attempt 1](https://github.com/antgroup/vsag/actions/runs/33587404487) |
| 2026-09-02T12:02:09.000Z | main (1.1) | [`dd67ab0`](https://github.com/antgroup/vsag/commit/dd67ab0700ed99abc222c5c2ef7382bff37c30d7) | Lint | [run #33624562699 attempt 1](https://github.com/antgroup/vsag/actions/runs/33624562699) |
| 2026-09-03T03:17:35.000Z | main (1.1) | [`56d87aa`](https://github.com/antgroup/vsag/commit/56d87aa40037f4498090ce2206f372a51fcebce0) | Lint | [run #33708412907 attempt 1](https://github.com/antgroup/vsag/actions/runs/33708412907) |
| 2026-09-03T03:24:32.000Z | main (1.1) | [`95cae43`](https://github.com/antgroup/vsag/commit/95cae435f3b07bc61e0e441b494cbf89f67865e0) | Lint | [run #33708451436 attempt 1](https://github.com/antgroup/vsag/actions/runs/33708451436) |
| 2026-09-04T06:48:31.000Z | main (1.1) | [`99db578`](https://github.com/antgroup/vsag/commit/99db5783c9bf1d61605148f39a0e1ee5ae91d59f) | Lint | [run #33843675013 attempt 1](https://github.com/antgroup/vsag/actions/runs/33843675013) |
| 2026-09-07T02:55:55.000Z | main (1.1) | [`e2cb48e`](https://github.com/antgroup/vsag/commit/e2cb48e5bd028be8332cb3e4c78b5a311778e04c) | Lint | [run #34076014926 attempt 1](https://github.com/antgroup/vsag/actions/runs/34076014926) |
| 2026-09-07T03:33:36.000Z | main (1.1) | [`51545fb`](https://github.com/antgroup/vsag/commit/51545fbbcd85bfc0e03eaa7adc28c901678688ba) | Lint | [run #34077182716 attempt 1](https://github.com/antgroup/vsag/actions/runs/34077182716) |
| 2026-09-07T07:28:40.000Z | main (1.1) | [`1adf73c`](https://github.com/antgroup/vsag/commit/1adf73c40f52c448ff4d6d79585256d12a5b96f0) | Lint | [run #34092991451 attempt 1](https://github.com/antgroup/vsag/actions/runs/34092991451) |
| 2026-09-07T09:27:48.000Z | main (1.1) | [`b23c105`](https://github.com/antgroup/vsag/commit/b23c105e2d3a45d92f1c78da424d6868f347e896) | Lint | [run #34102114098 attempt 1](https://github.com/antgroup/vsag/actions/runs/34102114098) |

<!-- fingerprints: 0b41dd4274fdfaeab35a4bc77145e05157a1b051ab18afd775890ed6b859c614,2109d58a52161616dc2efcac18a122ca301774024fe2f50e5375f9c5b773bff3,2a65e0b115b2344e0321709c577e92142d7ed2b04c860af4aed25d540abcf952,62106c4a7d96f155caf785f05e57a78725e67f7183e8864a5a978b61ed95b4e8,8574095bc746d2a9f94dfa9a724fef0aa54c924c394108667cc4f5148e1702b1,b581d6241d1ecc39c1475c6c45d7bb40e4b60dff5c9b85fe73bfdd7a547038b8,caa1edd4bd10b3da34fc8a2b54cbcbe48409ed401873a6481ede0dd1aa67b5b4,e9a55e2696f8fae4f548fb95d9ad8300c2ef010e1a67dcc33046769f0fc69d9b,fe4ff507ab5c41b8f4f2f009edebf40708d669ff90042f999c0aaef66982bc89 -->
<!-- vsag-github-agent:ci-failure:end -->

--- a/src/factory/factory.cpp
+++ b/src/factory/factory.cpp
@@ -65,14 +65,12 @@ namespace {
 
 IndexCommonParam
 make_streaming_common_param(const MetadataPtr& metadata, Allocator* allocator) {
-    auto resource = allocator == nullptr
-                        ? std::make_shared<Resource>(Engine::CreateDefaultAllocator(), nullptr)
-                        : std::make_shared<Resource>(allocator, nullptr);
+    auto resource = allocator == nullptr ? Resource(Engine::CreateDefaultAllocator(), nullptr)
+                                         : Resource(allocator, nullptr);
 
     IndexCommonParam common_param;
-    common_param.allocator_ = resource->GetAllocator();
-    common_param.thread_pool_ =
-        std::dynamic_pointer_cast<SafeThreadPool>(resource->GetThreadPool());
+    common_param.allocator_ = resource.GetAllocator();
+    common_param.thread_pool_ = std::dynamic_pointer_cast<SafeThreadPool>(resource.GetThreadPool());
 
     auto basic_info = metadata->Get("basic_info");
     if (basic_info.Contains("dim")) {

--- a/src/storage/serialization.cpp
+++ b/src/storage/serialization.cpp
@@ -123,7 +123,10 @@ StreamHeader::ReadRaw(StreamReader& reader) {
         throw VsagException(ErrorType::INVALID_BINARY,
                             "streaming serialization metadata must be a JSON object");
     }
-    return {std::make_shared<Metadata>(metadata_json), std::move(metadata_string)};
+    StreamHeaderData result;
+    result.metadata = std::make_shared<Metadata>(std::move(metadata_json));
+    result.metadata_string = std::move(metadata_string);
+    return result;
 }
 
 FooterPtr

